tasks/Task.cpp

```diff
@@ -283,14 +300,37 @@ void Task::setGain()
 void Task::writeVelocities(vpColVector v)
 {
     base::LinearAngular6DCommand vel;
-    vel.linear[0] = v[0];
-    vel.linear[1] = v[1];
-    vel.linear[2] = v[2];
-    vel.angular[0] = v[3];
-    vel.angular[1] = v[4];
-    vel.angular[2] = v[5];
-    vel.time = base::Time::now();
+    visp::saturationValues saturation = _saturation.get();
+
+    for (int i=0; i < 3; ++i)
+    {
+        // write NaN on the non-desirable outputs in order to
+        // keep the consistence with the other controllers
+        // linear
+        if (!expected_inputs.linear[i])
+        {
+            v[i] = base::NaN<double>();
```

Comment:
Is there really a reason why you first update the `v` array and only then write in `vel` ? You could directly write `vel` ... (which would arguably be more readable ...)